### PR TITLE
feat: implement vendor and provider constructs

### LINKED ISSUE #746: Introduce vendor and provider constructs
This issue is part of mergestat/internal#127 to add a framework to support multiple Git hosting providers. Refer to [Bitbucket](https://www.notion.so/mergestat/Bitbucket-ab38d2a332674169b6973c51865c46e9) on Notion for more details around `vendor` and `provider` constructs.

  - [x]  Add `vendor` and migrate all dependent relations to use it, with the default vendor set to `gtihub`.
  - [x]  Add `provider` and seed with the `GitHub Cloud` integration as that is the only one we support at this moment.
  - [x]  Migrate authentication to follow the pattern outlined in [Bitbucket](https://www.notion.so/Bitbucket-ab38d2a332674169b6973c51865c46e9)
      - [x]  Refactor to support multiple active tokens for a given provider at any point
      - [x]  Support for marking a token / credential as default for the provider
      - [x]  Support revoking / deleting token
  - [x]  Migrate repository and sync types to follow the pattern outlined in [Bitbucket](https://www.notion.so/Bitbucket-ab38d2a332674169b6973c51865c46e9)

### PR DESCRIPTION
**Pull request by riyaz-ali**
This PR adds `vendor` and `provider` constructs as outlined in documents linked in #746. Closes #746

### CODE REVIEW

File: graphql/service-credentials/index.ts [hunk 1]
```diff
@@ -8,14 +8,16 @@ const { ENCRYPTION_SECRET } = process.env
...
+  provider: string
+  type: string
+  token: string
 }
 
 module.exports = makeExtendSchemaPlugin({
   typeDefs: gql`
     extend type Mutation {
-      replaceGitHubPAT(pat: String!): Boolean
+      add_token(provider: UUID!, type: String!, token: String!): Boolean
```
Comment (patrickdevivo) [inline graphql/service-credentials/index.ts:20]: we should use camelCase here just to be consistent with the rest of the GraphQL API
```suggestion
      addToken(provider: UUID!, type: String!, token: String!): Boolean
```

**Comment by riyaz-ali** [comment]
This also addresses #747

File: migrations/900000000000023_add_vendor_and_provider.up.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
+-- SQL migration to add support for vendor and provider constructs
+BEGIN;
+
+-- Table mergestat.vendor contains a list of all third-party vendors that mergestat supports.
+-- It is almost always read-only for users, and data can only be inserted in this table using
+-- a migration.
+CREATE TABLE mergestat.vendors (
+    name TEXT PRIMARY KEY,     -- the enumerated name of the vendor; try to keep it lower-case
+    displayName TEXT NOT NULL, -- the name shown on the user interface
```
Comment (patrickdevivo) [inline migrations/900000000000023_add_vendor_and_provider.up.sql:9]:
```suggestion
    display_name TEXT NOT NULL, -- the name shown on the user interface
```

to be consistent with our other column names

Comment (riyaz-ali) [reply migrations/900000000000023_add_vendor_and_provider.up.sql:9]: I've made the change in #808 (see [this commit](https://github.com/mergestat/mergestat/pull/808/commits/3d6c3cbb1da7f05a43afe8777c03110bc69ac001))

**Pull request merged**